## Maximum Circumferential Stress

The first step is to determine $K_I$ and $K_{II}$ for the given panel. Since this is a center crack in an infinite panel, we have

In [13]:
import numpy as np

tau = 1.0
sigma = 4.0
K_IC = 60.0 #ksi sqrt(in)
a = 1.5/2 #in

K_I = sigma*np.sqrt(np.pi*a)
K_II = tau*np.sqrt(np.pi*a)

#import a solver library to solve non-linear trig equations
from scipy import optimize

def myeqn(theta):
    return K_I*np.sin(theta) + K_II*(3*np.cos(theta)-1)

sol = optimize.root(myeqn,0)
theta_p = sol.x[0]
print theta_p*180/np.pi

-25.332938613


We find (using equation 11.14) that the direction of crack extension, $\theta_P = -25.3^\circ$. Next we need to find the fracture stress (11.15).
$$4K_{IC} = K_I\left(3\cos \frac{\theta_P}{2} + \cos \frac{3\theta_P}{2}\right) - 3K_{II}\left(\sin \frac{\theta_P}{2} + \sin \frac{3\theta_P}{2}\right) $$

In [14]:
3*np.cos(theta_p/2)+np.cos(3*theta_p/2)

3.7150062221650302

In [15]:
np.sin(theta_p/2)+np.sin(3*theta_p/2)

-0.83492859561943289

$$4(60) = \sigma_c \sqrt{\pi a} (3.715) + 3\tau_c \sqrt{\pi a} (0.835)$$
Since we know that $\tau_c = \frac{1}{4}\sigma_c$ we can substitute and solve
$$240 = \sigma_c \sqrt{.75 \pi} (3.715) + 3/4\sigma_c \sqrt{.75 \pi} (0.835)$$

In [16]:
s_c = 240/(np.sqrt(np.pi*a)*(3.715+.75*.835))
print s_c

36.015618499


We find that the critical stress in this case is $\sigma_c = 36.0 \text{ ksi}$

## Principal Stress Criterion

We can also use the principal stress criterion for the same conditions to compare the two methods.

In [17]:
theta_p = np.arctan(-0.5)/2
theta_p*180/np.pi

-13.282525588538995

Now that we have found the angle we can substitute into 11.17b

In [20]:
s_theta = 0*np.sin(theta_p)**2 + 1.0*np.cos(theta_p)**2 - 0.5*np.sin(theta_p)*np.cos(theta_p)
print s_theta

1.05901699437


We know solve
$$K_{IC} = 1.06 \sigma_c \sqrt{\pi a}$$

In [21]:
K_IC/(s_theta*np.sqrt(np.pi*a))

36.909890171596551

And we find that $\sigma_c = 36.9 \text{ ksi}$